### ISSUE TITLE
Jenkins Pipeline 'tool' step not installing Terraform automatically (1.0.8)

### ISSUE THREAD

**Issue by danielbcorreia** [dpires/jenkins-terraform-plugin#6]
After configuring the terraform tool, the resulting XML is the following:

```xml
<?xml version='1.0' encoding='UTF-8'?>
<org.jenkinsci.plugins.terraform.TerraformBuildWrapper_-DescriptorImpl plugin="terraform@1.0.8">
 <installations>
   <org.jenkinsci.plugins.terraform.TerraformInstallation>
     <name>Terraform 0.7.8</name>
     <home>/var/jenkins_home//terraform_0.7.8_linux_amd64</home>
     <properties>
       <hudson.tools.InstallSourceProperty>
         <installers>
           <org.jenkinsci.plugins.terraform.TerraformInstaller>
             <id>0.7.8-linux-amd64</id>
           </org.jenkinsci.plugins.terraform.TerraformInstaller>
         </installers>
       </hudson.tools.InstallSourceProperty>
     </properties>
   </org.jenkinsci.plugins.terraform.TerraformInstallation>
 </installations>
</org.jenkinsci.plugins.terraform.TerraformBuildWrapper_-DescriptorImpl>
```

This seems fine, however when using the "tool" pipeline step the terraform binary is not installed automatically.

I've also tried creating a freestyle job and selecting the terraform option on the "Build Environment" section, and that works ok.

How can we make this work with the jenkins pipelines?

**Comment by dpires** [comment]
I haven't looked into the pipeline tool API. Is there another plugin you are using which installs itself when using a pipeline? If so, I can look into how to get this plugin to do the same.

**Comment by dpires** [comment]
I have released a fix for this in 1.0.10. 

The Terraform tool you reference in the pipeline with tool 'toolname'  will now be automatically installed if it isn't already.
